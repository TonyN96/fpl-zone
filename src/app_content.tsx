import React from "react";
import { useQuery } from "react-query";
import { BrowserRouter as Router,Route, Routes } from "react-router-dom";
import { useMediaQuery } from "@mui/material";
import { getAllFixtures, getGameData } from "api/fpl_api_provider";
import {
  AccountPage,
  AnalysisPage,
  FixturesAndResultsPage,
  GameweekLivePage,
  LoginPage,
  MyTeamPage,
  RegisterPage,
  ResetPage
} from "pages";
import PrivateRoute from "private_route";
import { AppData } from "types";

import { Logout } from "components/authentication";
import { Notifier, Startup } from "components/layout";

export const AppDataContext = React.createContext<AppData | null>(null);

export default function AppContent(): JSX.Element {
  // Fetching game data which will be made available throughout the app via context provider
  const {
    data: gameData,
    isError: gameDataIsError,
    error: gameDataError,
    isLoading: gameDataIsLoading
  } = useQuery("game-data", getGameData);

  // Fetching fixture data which will be made available throughout the app via context provider
  const {
    data: fixtureData,
    isError: fixtureDataIsError,
    error: fixtureDataError,
    isLoading: fixtureDataIsLoading
  } = useQuery("all-fixtures", getAllFixtures);

  const isCompact = useMediaQuery("(max-width:1500px)");

  const isLoading = gameDataIsLoading || fixtureDataIsLoading;
  const isError = gameDataIsError || fixtureDataIsError;
  const appData = gameData && fixtureData ? { ...gameData, fixtures: fixtureData, isCompact } : null;

  if (isLoading) {
    // Display loading message if data is still being fetched
    return (
      <Startup>
        <Notifier />
      </Startup>
    );
  } else if (isError) {
    // Display error message if data fetch failed
    const error = gameDataError || fixtureDataError;
    const errorMessage = error instanceof Error ? `: ${error.message}` : ".";

    return (
      <Startup>
        <Notifier type='error' message={`An error has occured: ${errorMessage}`} />
      </Startup>
    );
  }

  // Otherwise, render the app
  return (
    <AppDataContext.Provider value={appData}>
      <Router>
        <Routes>
          <Route path='*' element={<LoginPage />} />
          <Route path='/' element={<LoginPage />} />
          <Route path='/login' element={<LoginPage />} />
          <Route path='/register' element={<RegisterPage />} />
          <Route path='/reset' element={<ResetPage />} />
          <Route path='/logout' element={<Logout />} />
          <Route path='/account' element={<PrivateRoute component={<AccountPage />} />} />
          <Route
            path='/gameweek-live'
            element={<PrivateRoute component={<GameweekLivePage />} />}
          />
          <Route path='/my-team' element={<PrivateRoute component={<MyTeamPage />} />} />
          <Route
            path='/fixtures-and-results'
            element={<PrivateRoute component={<FixturesAndResultsPage />} />}
          />
          <Route path='/analysis' element={<PrivateRoute component={<AnalysisPage />} />} />
        </Routes>
      </Router>
    </AppDataContext.Provider>
  );

}
