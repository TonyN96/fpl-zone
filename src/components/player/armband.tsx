import React from "react";
import { Box, Typography } from "@mui/material";

interface ArmbandProps {
  isVice?: boolean;
}

export const Armband = ({ isVice = false }: ArmbandProps): JSX.Element => {
  return (
    <Box
      bgcolor='rgb(240, 240, 240, 1)'
      border='1px solid black'
      borderRadius='50%'
      className='flex-center'
      data-testid='armband-container'
      height='1.8vw'
      left={0}
      position='absolute'
      top={0}
      width='1.8vw'
    >
      <Typography>{isVice ? "V" : "C"}</Typography>
    </Box>
  );
};
