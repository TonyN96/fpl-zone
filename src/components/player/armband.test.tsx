import React from "react";
import { render, screen } from "@testing-library/react";
import { AppDataContext } from "app_content";
import { mockAppData } from "test";

import "@testing-library/jest-dom/extend-expect";

import { Armband } from "./armband";

describe("Armband tests", () => {
  const createComponent = (isVice: boolean): JSX.Element => {
    return (
      <AppDataContext.Provider value={mockAppData}>
        <Armband isVice={isVice} />
      </AppDataContext.Provider>
    );
  };

  describe("Text content displays as expected", () => {
    it("when captain", () => {
      render(createComponent(false));

      const container = screen.getByTestId("armband-container");

      expect(container).toHaveTextContent("C");
      expect(container).not.toHaveTextContent("V");
    });

    it("when vice captain", () => {
      render(createComponent(true));

      const container = screen.getByTestId("armband-container");

      expect(container).toHaveTextContent("V");
      expect(container).not.toHaveTextContent("C");
    });
  });
});
