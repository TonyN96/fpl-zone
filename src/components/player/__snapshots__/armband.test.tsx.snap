// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Armband tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1jev9c2"
    data-testid="armband-container"
  >
    C
  </div>
</DocumentFragment>
`;
